packages/template-retail-react-app/app/hooks/use-datacloud.js

```diff
@@ -396,10 +378,23 @@ export class DataCloudApi {
             ]
         }
 
-        try {
-            this.sdk.webEventsAppSourceIdPost(interaction)
-        } catch (err) {
-            logger.error('Error sending DataCloud event', err)
+        this.sdk.webEventsAppSourceIdPost(interaction)
+            .catch((err) => this._handleApiError(err))
+    }
+
+    _handleApiError(err) {
+        if (err?.response?.status === 400) {
+            logger.warn(
+                '[DataCloudApi] 400 Bad Request: Check your Data Cloud configuration (appSourceId, tenantId) and event payload.',
+                {
+                    url: err?.response?.url,
+                    status: err?.response?.status,
+                    statusText: err?.response?.statusText,
+                    data: err?.response?.data // if available
+                }
+            )
+        } else {
+            logger.error('[DataCloudApi] Error sending Data Cloud event', err)
```

Comment:
Please check how `logger` is called in other files. It expects certain parameter format.